lib/model/model.go

```diff
@@ -1465,31 +1472,50 @@ func (m *Model) numHashers(folder string) int {
 // clusterConfig returns a ClusterConfigMessage that is correct for the given peer device
 func (m *Model) clusterConfig(device protocol.DeviceID) protocol.ClusterConfigMessage {
 	cm := protocol.ClusterConfigMessage{
+		DeviceName:    m.deviceName,
 		ClientName:    m.clientName,
 		ClientVersion: m.clientVersion,
-		Options: []protocol.Option{
-			{
-				Key:   "name",
-				Value: m.deviceName,
-			},
-		},
 	}
 
 	m.fmut.RLock()
 	for _, folder := range m.deviceFolders[device] {
+		folderCfg := m.cfg.Folders()[folder]
 		cr := protocol.Folder{
-			ID: folder,
+			ID:              folder,
+			RescanIntervalS: uint32(folderCfg.RescanIntervalS),
+			Order:           uint32(folderCfg.Order),
```

Comment:
We're locking the various Order constants into the protocol here, so they should probably live in the protocol package from this point on?